File: hw/maisincho/u3/HW18Singleton/SingletonPy/model/USTax.py
class USTax:
    __instance = None  # Static private variable to hold the single instance

    def __init__(self, tax_percentage):
        if USTax.__instance is not None:
            raise Exception("This class is a singleton! Use get_instance() instead.")
        self.__tax_percentage = tax_percentage
        USTax.__instance = self

    @classmethod
    def get_instance(cls):
        if cls.__instance is None:
            cls.__instance = USTax(0.2)
        return cls.__instance

    @classmethod
    def set_instance(cls, new_instance):
        if new_instance is not None:
            cls.__instance = new_instance
        else:
            print("Cannot set null instance")

    def get_tax_percentage(self):
        return self.__tax_percentage

    def set_tax_percentage(self, tax_percentage):
        self.__tax_percentage = tax_percentage

File: hw/andrade/u3/HW18Singleton/HW18Singleton/SingletonPattern/src/main/java/ec/edu/espe/singletonpattern/model/USTax.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.singletonpattern.model;

/**
 *
 * @author Micaela Jácome
 */
public class USTax {

    private static USTax instance;
    private float taxPercentage;

    public USTax(float taxPercentage) {
        this.taxPercentage = taxPercentage;
    }

    public static USTax getInstance() {
        if (instance == null) {
            instance = new USTax(0.2F);
        }
        return instance;

    }

    public static void setInstance(USTax newInstance) {
        if (newInstance != null) {
            instance = newInstance;
        } else {
            System.out.println("Cannot set null instance ");
        }
    }

    public float getTaxPercentage() {
        return taxPercentage;
    }

    public void setTaxPercentage(float taxPercentage) {
        this.taxPercentage = taxPercentage;
    }
}
